transaction-service-api/src/main/java/pl/adamboguszewski/transactionservice/service/api/transaction/CreateTransactionRequest.java

```diff
@@ -0,0 +1,55 @@
+package pl.adamboguszewski.transactionservice.service.api.transaction;
+
+import lombok.Value;
+import lombok.NonNull;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+@Value
+public class CreateTransactionRequest {
+
+    @NonNull
+    Long totalPrice;
+
+    @NonNull
+    List<Product> products;
+
+    @NonNull
+    Info transactionInfo;
+
+    @Value
+    public static class Info {
+
+        @NonNull
+        LocalDateTime transactionTime;
+
+        @NonNull
+        UUID checkoutId;
+
+        @NonNull
+        List<PaymentInfo> paymentsInfo;
+
+        @Value
+        public static class PaymentInfo {
+
+            @NonNull
+            Long amount;
+
+            @NonNull
+            Long multiplier;
+        }
+    }
+
+    @Value
+    public static class Product {
+
+        @NonNull
+        UUID productId;
+
+        @NonNull
+        Long quantity;
```

Comment:
Elon: Son go to shop and buy 10* ced37c14-d18c-11ea-87d0-0242ac130003's 
X Æ A-Xii: Okay dad